pkg/apiserver/installer/cmhandlers.go

```diff
@@ -64,6 +64,20 @@ func (ch *CMHandlers) registerResourceHandlers(a *MetricsAPIInstaller, ws *restf
 		return err
 	}
 
+	listOptions, _, _ := lister.NewListOptions()
```

Comment:
please add a comment clarifying why we have both list options and extra list options

Reply:
As the standard `ListOptions` is defined as a struct and not an interface, I don't know of a way to use a single object that could be extended with polymorphism. So there is a standard list options as defined by the `ListOptions` struct, plus the extra options defined by the CM adapter.

I'll add a comment in the code when the question where the `ListerWithOptions` should go is resolved.

Reply:
My initial suggestion would be to say that custom list options should `,inline` embed the base ListOptions, but I think your idea is actually fine as well :-).